### DISCUSSION TITLE
Problema al correr test de parte 2

### DISCUSSION THREAD

**Discussion by EAOT** [IIC2133-PUC/2024-2#69] [Tarea 0] [answered]
Al momento de correr los test de la parte 2, me pasa que al correr mi codigo los eventos TRADE y GIVE luego por alguna razon siempre obtengo un segmentation fault y he buscado la fuente del error pero no consigo lograrlo, ademas me di cuenta que al comentar las acciones a realizar por mi codigo en esos comandos el problema desaparece pero en realidad necesito ayuda.

**Comment by waknaudt** [comment]
Te recomendaria aprender a usar `gdb` para debuggear C, a mi me ayudo muchisimo para encontrar las causas de los Segmentation fault.

**Reply by FabriRandon** [reply]
Lo que me sirvió a mi fue crear un script que me permita comparar los outputs (el de la pauta y el que obtuve) de manera de detectar en dónde fue el error. De esta manera logré corregir todos los errores super rápido ya que muchos outputs tienen como 1 millón de lineas.

**Reply by FabriRandon** [reply]
Bueno y también usar los comandos de depuración en conjunto

**Comment by sebacarrascop** [comment, marked as answer]
hola!, complementando a lo que te dicen, siempre puedes usar un test simple (o crear uno de cero) y añadirle la función que te da problemas, luego será mucho más facil de debuggear si tienes claro como debería comportarse tu código, saludos.
